internal/saucecloud/cloud.go

```diff
@@ -423,102 +458,78 @@ func checkPathLength(projectFolder string, matcher sauceignore.Matcher) (string,
 	return "", nil
 }
 
-// archiveFolder creates a zip file in the tempDir directory from the given projectFolder.
-func (r *CloudRunner) archiveFolder(project interface{}, tempDir string, projectFolder string, sauceignoreFile string) (string, error) {
-	matcher, err := r.getFileMatcher(projectFolder, sauceignoreFile)
-	if err != nil {
-		return "", err
-	}
+// archiveNodeModules archives node_modules located under rootDir and returns the path to the zip file. Returns an empty
+// string if node_modules doesn't exist or is actively ignored.
+func (r *CloudRunner) archiveNodeModules(tempDir string, rootDir string, matcher sauceignore.Matcher) (string, error) {
+	modDir := filepath.Join(rootDir, "node_modules")
+	ignored := matcher.Match([]string{modDir}, true)
 
-	if path, err := checkPathLength(projectFolder, matcher); err != nil {
-		msg.PathTooLongForArchive(path)
-		return "", err
-	}
-
-	start := time.Now()
+	_, err := os.Stat(modDir)
+	hasMods := err == nil
+	wantMods := len(r.NPMDependencies) > 0
 
-	zipName := filepath.Join(tempDir, "app.zip")
-	z, err := zip.NewFileWriter(zipName, matcher)
-	if err != nil {
-		return "", err
+	if !hasMods && wantMods {
+		return "", fmt.Errorf("unable to access 'node_modules' folder, but you have npm dependencies defined in your project")
```

Comment:
"project" in this context might be confusing? 

```suggestion
		return "", fmt.Errorf("unable to access 'node_modules' folder, but you have npm dependencies defined in your configuration")
```

And I like the suggested action in the next error message. Maybe the same here would be good too?